include/optimizer.h

```diff
@@ -24,8 +24,9 @@ class Optimizer {
   /*!
   * \brief Operator constructor, the optimizer is not initialized until the first Update
   * \param opt_type type of the optimizer
+  * \param learning_rate
   */
-  explicit Optimizer(const std::string &opt_type);
+  Optimizer(const std::string &opt_type, mx_float learning_rate);
```

Comment:
Do we really need to give a learning_rate at the initialization?
If so, the weight decay is also needed.
And I think it better to make learning rate and weight decay mutable during the update.